pkg/iprange/parse_test.go

```diff
@@ -0,0 +1,258 @@
+			input:   "192.0.2.0-192.0.2.",
+			wantErr: true,
+		},
+		"multiple invalid syntax": {
+			input:   "2001:db8::0 192.0.2.0-192.0.2.10 2001:db8::0/999 192.0.2.0/255.255.255.0",
+			wantErr: true,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			rs, err := ParseRanges(test.input)
+
+			if test.wantErr {
+				assert.Error(t, err)
+				assert.Nilf(t, rs, "want: nil, got: %s", rs)
+			} else {
+				assert.NoError(t, err)
+				assert.Equalf(t, test.wantRanges, rs, "want: %s, got: %s", test.wantRanges, rs)
+			}
+		})
+	}
+}
+
+func TestParseRange(t *testing.T) {
+	tests := map[string]struct {
+		input     string
+		wantRange Range
+		wantErr   bool
+	}{
+		"v4 IP": {
+			input:     "192.0.2.0",
+			wantRange: prepareV4Range("192.0.2.0", "192.0.2.0"),
+		},
+		"v4 IP: invalid address": {
+			input:   "192.0.2.",
+			wantErr: true,
+		},
+		"v4 Range": {
+			input:     "192.0.2.0-192.0.2.10",
+			wantRange: prepareV4Range("192.0.2.0", "192.0.2.10"),
+		},
+		"v4 Range: start == end": {
+			input:     "192.0.2.0-192.0.2.0",
+			wantRange: prepareV4Range("192.0.2.0", "192.0.2.0"),
+		},
+		"v4 Range: start > end": {
+			input:   "192.0.2.10-192.0.2.0",
+			wantErr: true,
+		},
+		"v4 Range: invalid start": {
+			input:   "192.0.2.-192.0.2.10",
+			wantErr: true,
+		},
+		"v4 Range: invalid end": {
+			input:   "192.0.2.0-192.0.2.",
+			wantErr: true,
+		},
+		"v4 Range: v6 start": {
+			input:   "2001:db8::0-192.0.2.10",
+			wantErr: true,
+		},
+		"v4 Range: v6 end": {
+			input:   "192.0.2.0-2001:db8::0",
+			wantErr: true,
+		},
+		"v4 CIDR: /0": {
+			input:     "192.0.2.0/0",
+			wantRange: prepareV4Range("0.0.0.1", "255.255.255.254"),
+		},
+		"v4 CIDR: /24": {
+			input:     "192.0.2.0/24",
+			wantRange: prepareV4Range("192.0.2.1", "192.0.2.254"),
+		},
+		"v4 CIDR: /30": {
+			input:     "192.0.2.0/30",
+			wantRange: prepareV4Range("192.0.2.1", "192.0.2.2"),
+		},
+		"v4 CIDR: /31": {
+			input:     "192.0.2.0/31",
+			wantRange: prepareV4Range("192.0.2.0", "192.0.2.1"),
+		},
+		"v4 CIDR: /32": {
+			input:     "192.0.2.0/32",
+			wantRange: prepareV4Range("192.0.2.0", "192.0.2.0"),
+		},
+		"v4 CIDR: missing prefix length": {
+			input:   "192.0.2.0/",
+			wantErr: true,
+		},
+		"v4 CIDR: invalid prefix length": {
+			input:   "192.0.2.0/99",
+			wantErr: true,
+		},
+		"v4 Mask: /0": {
+			input:     "192.0.2.0/0.0.0.0",
+			wantRange: prepareV4Range("0.0.0.1", "255.255.255.254"),
+		},
+		"v4 Mask: /24": {
+			input:     "192.0.2.0/255.255.255.0",
+			wantRange: prepareV4Range("192.0.2.1", "192.0.2.254"),
+		},
+		"v4 Mask: /30": {
+			input:     "192.0.2.0/255.255.255.252",
+			wantRange: prepareV4Range("192.0.2.1", "192.0.2.2"),
+		},
+		"v4 Mask: /31": {
+			input:     "192.0.2.0/255.255.255.254",
+			wantRange: prepareV4Range("192.0.2.0", "192.0.2.1"),
+		},
+		"v4 Mask: /32": {
+			input:     "192.0.2.0/255.255.255.255",
+			wantRange: prepareV4Range("192.0.2.0", "192.0.2.0"),
+		},
```

Comment:
Please, add here an inverted IP/MASK pair like `255.255.255.254/192.0.2.0`

Reply:
it is not inverted mask, it is not canonical form mask (ones followed by zeros)

Reply:
this is tested